include/lmms_math.h

```diff
@@ -135,82 +81,41 @@ static inline int fast_rand()
 	return( (unsigned)( next / 65536 ) % 32768 );
 }
 
-static inline double fastRand( double range )
-{
-	static const double fast_rand_ratio = 1.0 / FAST_RAND_MAX;
-	return fast_rand() * range * fast_rand_ratio;
-}
-
 static inline float fastRandf( float range )
 {
 	static const float fast_rand_ratio = 1.0f / FAST_RAND_MAX;
 	return fast_rand() * range * fast_rand_ratio;
 }
 
-//! @brief Takes advantage of fmal() function if present in hardware
-static inline long double fastFmal( long double a, long double b, long double c ) 
-{
-#ifdef FP_FAST_FMAL
-	#ifdef __clang__
-		return fma( a, b, c );
-	#else
-		return fmal( a, b, c );
-	#endif
-#else
-	return a * b + c;
-#endif // FP_FAST_FMAL
-}
 
-//! @brief Takes advantage of fmaf() function if present in hardware
-static inline float fastFmaf( float a, float b, float c ) 
+//! Round `value` to `where` depending on step size
+template<class T>
+static void roundAt(T& value, const T& where, const T& stepSize)
 {
-#ifdef FP_FAST_FMAF
-	#ifdef __clang__
-		return fma( a, b, c );
-	#else
-		return fmaf( a, b, c );
-	#endif
-#else
-	return a * b + c;
-#endif // FP_FAST_FMAF
+	if (std::abs(value - where) < F_EPSILON * std::abs(stepSize))
+	{
+		value = where;
+	}
 }
 
-//! @brief Takes advantage of fma() function if present in hardware
-static inline double fastFma( double a, double b, double c ) 
-{
-#ifdef FP_FAST_FMA
-	return fma( a, b, c );
-#else
-	return a * b + c;
-#endif
-}
 
-// source: http://martin.ankerl.com/2007/10/04/optimized-pow-approximation-for-java-and-c-c/
-static inline double fastPow( double a, double b )
-{
-	union
-	{
-		double d;
-		int32_t x[2];
-	} u = { a };
-	u.x[1] = static_cast<int32_t>( b * ( u.x[1] - 1072632447 ) + 1072632447 );
-	u.x[0] = 0;
-	return u.d;
+//! returns 1.0f if val >= 0.0f, -1.0 else
+static inline float sign(float val) 
+{ 
+	return val >= 0.0f ? 1.0f : -1.0f; 
 }
 
-// sinc function
-static inline double sinc( double _x )
+
+//! if val >= 0.0f, returns sqrtf(val), else: -sqrtf(-val)
+static inline float sqrt_neg(float val) 
 {
-	return _x == 0.0 ? 1.0 : sin( F_PI * _x ) / ( F_PI * _x );
+	return sqrtf(std::abs(val)) * sign(val);
```

Comment:
Not a big problem, since the code was already like this, but why not change `sqrtf` to `std::sqrt` too? Also, `x * sign(y)` can be replaced with `std::copysign(x, y)` when `x` is guaranteed to be non-negative.
```suggestion
	return std::copysign(std::sqrt(std::abs(val)), val);
```

Reply:
I agree about the sqrt, but about `std::copysign`, I'm not sure.

Reply:
changed sqrt for now.